### DISCUSSION TITLE
Is it possible to get an OpenGL Core (non-ES) context?

### DISCUSSION THREAD

**Discussion by ianhattendorf** [slint-ui/slint#6482] [Q&A]
I'm using an approach similar to https://github.com/slint-ui/slint/tree/master/examples/opengl_underlay to render with an OpenGL context. This gets me an OpenGL ES 3.2 context, however I would like to use features from OpenGL 4.4+.

I've attempted `SLINT_BACKEND=winit-skia-opengl` and `SLINT_BACKEND=winit-femtovg` without any luck. Is this supported, or if not, would it be a desired feature?

This is on Fedora 40 Sway (Wayland) w/ an RX 5700 XT. I also occasionally test on a Windows 11 laptop with an Intel 9th gen UHD 630 + RTX 2000 GPU.

**Comment by tronical** [comment]
In general, this would be a desired feature to offer, but I think it has to be opt-in by API and is part of a bigger topic: Offering a kind of `BackendBuilder` API that allows the developer to select certain features they'd like the backend to provide, and then Slint (the backend selector) either creates and configures a suitable backend, or produces an error. (`Result`).

(pseudo code a bit with unnecessary type annotations to make it easier to understand though)

```rust
    let builder = slint::platform::PlatformBuilder::new().with_opengl_profile(slint::OpenGLProfile::Core);
    let result: Result<Box<dyn slint::Platform>>, slint::PlatformError> = builder.build();
    slint::set_platform(result?);
```

**Reply by ianhattendorf** [reply]
I like this suggestion. I opened https://github.com/slint-ui/slint/pull/6510 to continue the discussion.
